homeassistant/components/tautulli/__init__.py

```diff
@@ -61,7 +61,7 @@ def __init__(
         self.entity_description = description
         self.user = user
         self._attr_device_info = DeviceInfo(
-            configuration_url=coordinator.host_configuration.base_url,
+            configuration_url=str(coordinator.host_configuration.base_url),
```

Comment:
IMO this is a workaround.
I mentioned this as an option to @emontnemery a few weeks ago as well.
Is there a reason we can not use yarl.URL objects to represent URLs?
Is that not a better representation of it?

Reply:
configuration_url is defined/typed as `str | None`

![CleanShot 2023-07-27 at 09 51 22](https://github.com/home-assistant/core/assets/195327/fd5dcd40-551a-4291-b75d-370fa5aeb426)

The difference is that validation of the contents of DeviceInfo has been added. 

![CleanShot 2023-07-27 at 09 52 10](https://github.com/home-assistant/core/assets/195327/963070bd-4d4e-46bb-a040-37bfbcc4d996)


I do get your comment, as YARL does implement the `__str__` dunder, however, that is not really checkable?


I guess alternatively, we could cast anything to a string in the check and try to parse that...

Will take a look at that 👍

Reply:
If the value is already URL, there is no need to call urlparse on it, as it already provides a `scheme` attribute that could be validated directly.

Reply:
Oh, and I get (and know) that this does not comply with the current typing of it.
I was holding off on working around that while Erik explored options to see if we should allow it.